pkg/api/query/grpc.go

```diff
@@ -187,7 +200,16 @@ func (g *GRPCAPI) QueryRange(request *querypb.QueryRangeRequest, srv querypb.Que
 	endTime := time.Unix(request.EndTimeSeconds, 0)
 	interval := time.Duration(request.IntervalSeconds) * time.Second
 
-	qry, err := g.queryEngine.NewRangeQuery(queryable, &promql.QueryOpts{LookbackDelta: lookbackDelta}, request.Query, startTime, endTime, interval)
+	var engine v1.QueryEngine
+	switch promqlEngineType(request.Engine) {
+	case promqlEnginePrometheus:
+		engine = g.queryEngine
+	case promqlEngineThanos:
+		engine = g.thanosEngine
+	default:
+		engine = g.queryEngine
```

Comment:
I think erroring out here would be better, rather than just implicitly using an engine.